File: Backend/TopSys.TopConWeb.Domain/Entities/BombaPrecoTerceiro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TopSys.TopConWeb.Domain.Enums;
using TopSys.TopConWeb.Domain.Interfaces.Entities;

namespace TopSys.TopConWeb.Domain.Entities
{
    public class BombaPrecoTerceiro : IBombaPreco
    {
        public int BombistaCodigo { get; set; }
        public virtual Interveniente Bombista { get; set; }

        public int BombaTipoCodigo { get; set; }
        public CadastroGeral BombaTipo { get; set; }

        public DateTime DataInicioVigencia { get; set; }

        public int M3Ate { get; set; }

        public float TaxaMinimaPreco { get; set; }

        public float M3Preco { get; set; }

        public int M3AteValorMinimo { get; set; }

        public float TaxaMinimaPrecoPercentualDescontoMaximo { get; set; }
        public float TaxaMinimaPrecoValorMinimo { get; set; }

        public float M3PrecoPercentualDescontoMaximo { get; set; }
        public float M3PrecoValorMinimo { get; set; }

        public EBombaM3CalculoTipo TipoCalculo { get; set; }

        public float HoraAte { get; set; }

        public float HoraTaxaMinimaPreco { get; set; }

        public float HoraPreco { get; set; }

        public float HoraAteValorMinimo { get; set; }

        public float HoraTaxaMinimaPrecoValorMinimo { get; set; }

        public float HoraPrecoValorMinimo { get; set; }

        public EBombaHoraCalculoTipo HoraTipoCalculo { get; set; }
    }
}

File: Frontend/TopConWeb-AngularV8/src/app/classes/bomba/bomba-preco-terceiro.ts
import { Interveniente } from '../interveniente/interveniente';
import { CadastroGeral } from '../cadastro-geral/cadastro-geral';
import { EBombaM3CalculoTipo } from './bomba-preco';

export class BombaPrecoTerceiro {
    bombista: Interveniente;
    bombaTipo: CadastroGeral;
    dataInicioVigencia: Date;
    m3Ate: number = 0;
    taxaMinimaPreco: number = 0.0;
    m3Preco: number = 0.0;
    m3AteValorMinimo: number = 0;
    taxaMinimaPrecoPercentualDescontoMaximo: number = 0.0;
    taxaMinimaPrecoValorMinimo: number = 0.0;
    m3PrecoPercentualDescontoMaximo: number = 0.0;
    m3PrecoValorMinimo: number = 0.0;
    tipoCalculo: EBombaM3CalculoTipo = EBombaM3CalculoTipo.taxaMinimaOuExcedente;
}